src/core/server/saved_objects/export/saved_objects_exporter.ts

```diff
@@ -150,27 +161,113 @@ export class SavedObjectsExporter {
     return bulkGetResult.saved_objects;
   }
 
+  /**
+   * Generator which wraps calls to `SavedObjects.find` and iterates over
+   * multiple pages of results using `_pit` and `search_after`. This will
+   * open a new PIT, continue paging until a set of results is received
+   * that's smaller than the designated `perPage`, and then close the PIT.
+   *
+   * @example
+   * ```ts
+   * const finder = findWithPointInTime({
+   *   type: 'index-pattern',
+   *   search: 'foo*',
+   *   perPage: 100,
+   * });
+   *
+   * const responses: SavedObjectFindResponse[] = [];
+   * for await (const response of finder) {
+   *   responses.push(...response);
+   * }
+   * ```
+   */
+  private async *findWithPointInTime(findOptions: SavedObjectsFindOptions) {
```

Comment:
This could have been written a handful of ways, but my main goal was to create something where all of the PIT logic was self-contained. Open to any feedback here.

Reply:
> it feels like providing this as a utility function on the saved objects client could be valuable

++ Yeah I had this in mind when I was writing this, and it is part of the reason I structured it the way I did.

> the ES client will already retry failed requests 3 times before throwing

Great point about the retries, I will add some logic to close the PIT with a log message on failure.

Reply:
I went ahead and extracted the generator into a separate file to make it easier to expose as a utility, but haven't actually exposed it from the SO client yet. PIT closing should also now be happening on any errors.

Reply:
@rudolf Just to confirm we are on the same page -- as the update touches several files 🙂 -- you're proposing attaching this directly to the SO client, correct? Not as a standalone static helper. i.e. something like `savedObjectsClient.findWithPointInTime`

Reply:
I haven't given it that much thought, but on the client seems like the most useful and the most likely place developers will look for it. I'd suggest we expose it in a follow-up PR though since this one is already quite beefy.

Reply:
👍  Agreed, I'll add to the list of follow ups.